autotest/ogr/ogr_gpkg.py

```diff
@@ -4709,34 +4709,97 @@ def test_ogr_gpkg_creation_fid():
+    )
+    ds.ExecuteSQL("""CREATE TABLE "poly"("fid" INTEGER PRIMARY KEY, "geom" POLYGON)""")
+    ds = None
 
     gdal.ErrorReset()
     with gdaltest.error_handler():
-        ds = ogr.Open(tmpfile)
+        ds = ogr.Open(out_filename)
     assert ds.GetLayerCount() == 1, "bad layer count"
-    assert (
-        gdal.GetLastErrorMsg().find("Table poly appearing several times") >= 0
-    ), "should NOT have warned"
+    assert gdal.GetLastErrorMsg() != ""
     ds = None
 
-    gdal.Unlink(tmpfile)
+    gdal.Unlink(out_filename)
+
+
+###############################################################################
+# Test opening a non-standard GeoPackage with multiple geometry columns
+
+
+def test_ogr_gpkg_multiple_geom_columns():
+
+    out_filename = "/vsimem/test_ogr_gpkg_multiple_geom_columns.gpkg"
+    ogr.GetDriverByName("GPKG").CreateDataSource(out_filename)
+
+    ds = ogr.Open(out_filename, update=1)
+    ds.ExecuteSQL("DROP TABLE gpkg_geometry_columns")
+    # Modified gpkg_geometry_columns definition with a UNIQUE constraint on both (table_name, column_name)
+    ds.ExecuteSQL(
+        """CREATE TABLE gpkg_geometry_columns (table_name TEXT NOT NULL,column_name TEXT NOT NULL,geometry_type_name TEXT NOT NULL,srs_id INTEGER NOT NULL,z TINYINT NOT NULL,m TINYINT NOT NULL,CONSTRAINT pk_geom_cols PRIMARY KEY (table_name, column_name),CONSTRAINT uk_gc_table_name_column_name UNIQUE (table_name, column_name),CONSTRAINT fk_gc_tn FOREIGN KEY (table_name) REFERENCES gpkg_contents(table_name),CONSTRAINT fk_gc_srs FOREIGN KEY (srs_id) REFERENCES gpkg_spatial_ref_sys (srs_id));"""
+    )
+    ds.ExecuteSQL(
+        """CREATE TABLE "test" ( "ogc_fid" INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL, "poly" POLYGON, "pt" POINT, "area" REAL, "eas_id" INTEGER, "prfedea" TEXT(16))"""
+    )
+    ds.ExecuteSQL(
+        "INSERT INTO test VALUES(1,X'4750000300000000000000401f401d41000000e05e511d4100000080322c5241000000001d2d52410103000000010000001b000000000000c01a481d4100000080072d5241000000e0814b1d41000000001d2d524100000040c44b1d41000000000f2d5241000000002c4c1d41000000a0002d524100000000774d1d41000000c0072d5241000000a0c44e1d4100000080112d52410000002008501d41000000c0172d5241000000e05e511d4100000020dd2c5241000000405e511d4100000040cf2c524100000000f0501d41000000c0ba2c52410000008084501d4100000020af2c524100000040a94f1d4100000000a42c524100000080744e1d41000000a09a2c524100000040014f1d41000000c0852c5241000000e0e04d1d4100000020872c524100000040f8441d41000000e0432c5241000000c012441d4100000080322c524100000000ff431d4100000020362c5241000000004b431d4100000080552c52410000000030431d41000000c05d2c5241000000c09b421d4100000000712c524100000080d6411d4100000080912c52410000008027411d4100000040b22c5241000000401f401d4100000040d62c5241000000a043441d4100000060f02c524100000060aa461d4100000080ff2c5241000000c01a481d4100000080072d5241',X'4750000100000000010100000000000000804b1d41000000001d2d5241',NULL,170,NULL);"
```

Comment:
is it possible to use a human readable format for the geometry (WKT?) ?

Reply:
Unfortunately there's no ST_GeomFromText() for a GPKG datasource